include/albatross/src/covariance_functions/measurement.hpp

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (C) 2019 Swift Navigation Inc.
+ * Contact: Swift Navigation <dev@swiftnav.com>
+ *
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+#ifndef INCLUDE_ALBATROSS_SRC_COVARIANCE_FUNCTIONS_MEASUREMENT_HPP_
+#define INCLUDE_ALBATROSS_SRC_COVARIANCE_FUNCTIONS_MEASUREMENT_HPP_
+
+namespace albatross {
+
+template <typename X> struct Measurement {
+
+  Measurement(){};
+
+  Measurement(const X &x) { value = x; }
+
+  X value;
+};
+
+template <typename SubCovariance>
+class MeasurementOnly
+    : public CovarianceFunction<MeasurementOnly<SubCovariance>> {
+
+public:
+  MeasurementOnly(){};
+  MeasurementOnly(const SubCovariance &sub_cov) : sub_cov_(sub_cov){};
+
+  std::string name() const {
+    return "measurement[" + sub_cov_.get_name() + "]";
+  }
+
+  ParameterStore get_params() const override { return sub_cov_.get_params(); }
+
+  void unchecked_set_param(const ParameterKey &name,
+                           const Parameter &param) override {
+    sub_cov_.set_param(name, param);
+  }
+
+  /*
+   * This will create a scaled identity matrix, but only between
+   * two different observations defined by the Observed type.
+   */
+  template <
+      typename X, typename Y,
+      typename std::enable_if<
+          has_valid_call_impl<SubCovariance, X &, Y &>::value, int>::type = 0>
+  double _call_impl(const X &x, const Y &y) const {
+    return 0.;
```

Comment:
Does this match the comment above?

Reply:
Ha! not even close.  I'll delete it thanks.